transaction/core/src/input_rules.rs

```diff
@@ -18,13 +18,14 @@ use displaydoc::Display;
 use mc_crypto_digestible::Digestible;
 use prost::Message;
 use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
 
 /// A representation of rules on a transaction, imposed by the signer of some
 /// input in the transaction.
 ///
 /// Any rule could conceivably be added here if it can be evaluated against a
 /// `Tx`.
-#[derive(Clone, Digestible, PartialEq, Eq, Message, Serialize, Deserialize)]
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
```

Comment:
Consider adding `ZeroizeOnDrop` along with `Zeroize` (throughout this PR)
```suggestion
#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize, ZeroizeOnDrop)]
```
(requires `use`)

Reply:
Let's make a ticket to discuss this more, and do all of the `transaction-core` stuff using `zeroize(drop)`, if we want this. In this PR I would rather just follow existing patterns.

There are also tradeoffs here, because `Drop` types cannot be `Copy`.

I think there's may also be a world where objects like `Tx` are zeroize on drop, but lower level objects like `TxOut` are just zeroize. Not sure, needs more investigation.

None of this was possible until recently because `RistrettoPublic` did not implement Zeroize.

Reply:
This is now https://github.com/mobilecoinfoundation/mobilecoin/issues/2717